source/PlayServicesResolver/src/JavaUtilities.cs

```diff
@@ -58,20 +58,19 @@ internal class ToolNotFoundException : ApplicationException {
                 // Unity 2019.x added installation of the JDK in the AndroidPlayer directory
                 // so fallback to searching for it there.
                 if (String.IsNullOrEmpty(javaHome) || EditorPrefs.GetBool("JdkUseEmbedded")) {
+#if UNITY_2019_3_OR_NEWER
```

Comment:
We precompile the resolver into a DLL so any conditional logic needs to be conditional at runtime.

If it's _really_ only possible to change behavior based upon Unity version (it is super brittle), then you can use [ExecutionEnvironment.VersionMajorMinor](https://github.com/googlesamples/unity-jar-resolver/blob/ecfc41a15f4be10e8b3c88aca4352c4c17e38a3e/source/VersionHandlerImpl/src/ExecutionEnvironment.cs#L48)

Looking at the code below, I think you can check for the `UnityEditor.Android.AndroidExternalToolsSettings.jdkRootPath` property with reflection to determine the behavior rather than using the Unity version.

For examples, of how we're building accessors with reflection see https://github.com/googlesamples/unity-jar-resolver/blob/ecfc41a15f4be10e8b3c88aca4352c4c17e38a3e/source/PlayServicesResolver/src/UnityCompat.cs

Reply:
Thanks for the review, I will try reflection.